python/ray/dag/compiled_dag_node.py

```diff
@@ -530,9 +530,29 @@ def __init__(self):
                 super().__init__(daemon=True)
                 self.in_teardown = False
 
-            def teardown(self):
+            def wait_teardown(self):
+                for ref in outer.worker_task_refs:
+                    try:
+                        ray.get(ref, timeout=10)
+                    except ray.exceptions.GetTimeoutError:
+                        logger.warn(
+                            "At least one actor in the DAG is still running 10s "
+                            "after teardown(). Teardown may hang."
+                        )
+                    except Exception:
+                        continue
+
+                    try:
+                        ray.get(ref)
+                    except Exception:
+                        continue
+
+            def teardown(self, wait):
                 if self.in_teardown:
+                    if wait:
```

Comment:
Do we need this?  Since we are already in teardown, isn't there supposed to be other caller that is waiting?

Reply:
It's to make sure that explicitly calling `.teardown()` will always wait until the DAG is torn down.